### PR TITLE
#4270 Re-design & re-order recipient inputs

### LINKED ISSUE #4270: "from" selector in compose not obvious -> align with Gmail behavior
This "from" only shows when the user has more than one sending address (another sending alias)

Gmail makes it pretty obvious that it's "from"

image

Our own UI is a lot less obvious, and can confuse the user:

image

Todo in this issue:
 - add `From` and `To` (and `Cc` / `Bcc`) to the left of the input fields
 - switch order of inputs: `From` should be above `To` and others
 - (maybe) `From` would not have to be visible in "collapsed" (unfocused) view. It's only needed in the expanded view, similar to what Gmail does

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR Re-designed & re-order recipient inputs

close #4270 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (Only UI changes. current test will fail if something wrong)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by ioanmo226** [comment]
Please don't review yet. I need to fix functionality as some ui tests fail.
I'll mark this as ready for review once done

**Comment by ioanmo226** [comment]
image
image
image

File: extension/chrome/elements/compose.ts [hunk 1]
```diff
@@ -114,8 +114,11 @@ export class ComposeView extends View {
     recipients: 'span.recipients',
     contacts: '#contacts',
     input_addresses_container_outer: '#input_addresses_container',
+    recipient_left_label: '.recipient-left-label',
+    input_container_from: '#input-container-from',
     input_addresses_container_inner: '#input_addresses_container > div:first',
     recipients_inputs: '#input_addresses_container input',
+    recipients_toogle_elements: '#input_subject, #input_text, #input_password',
```
Comment (rrrooommmaaa) [inline extension/chrome/elements/compose.ts:121]: typo -- should be `toggle`

File: extension/chrome/elements/compose-modules/compose-recipients-module.ts [hunk 2]
```diff
@@ -323,20 +324,17 @@ export class ComposeRecipientsModule extends ViewModule<ComposeView> {
...
-      this.view.S.cached('input_addresses_container_outer').addClass('invisible');
-      this.view.S.cached('recipients_placeholder').css('display', 'flex');
-      this.setEmailsPreview();
-      this.hideContacts();
-      this.view.sizeModule.setInputTextHeightManuallyIfNeeded();
+  public collapseInputsIfNeeded = async () => {
+    if (this.view.S.cached('input_addresses_container_outer').hasClass('invisible')) {
+      return;
     }
+    await Promise.all(this.addedRecipients.map(r => r.evaluating)); // Wait untill all recipients loaded.
```
Comment (rrrooommmaaa) [inline extension/chrome/elements/compose-modules/compose-recipients-module.ts:331]: typo? `until`?

Comment (ioanmo226) [reply extension/chrome/elements/compose-modules/compose-recipients-module.ts]: 👍

File: test/source/browser/controllable.ts [hunk 3]
```diff
@@ -248,7 +248,6 @@ abstract class ControllableBase {
   };
 
   public selectOption = async (selector: string, choice: string) => {
-    await this.waitAll(selector, { visible: true });
```
[the remark this answers is not in the document]
Comment (rrrooommmaaa) [reply test/source/browser/controllable.ts:251]: This means the existing tests become weaker? The user can't see the option while the tests can and signal no error?

[the remark this answers is not in the document]
Comment (ioanmo226) [reply test/source/browser/controllable.ts:251]: Weaker doesn't seem correct word but implemented right way.
Option not shown on UI and can be changed with ui test is not wrong way.
Of course, we can show `from` selector (by clicking recipient placeholder label) if needed and check if `from` selector is still there.
But I don't think that's worth it in ui test.

[the remark this answers is not in the document]
Comment (ioanmo226) [reply test/source/browser/controllable.ts:251]: Moreover, as `selectOption` is global function, this function can't know how to show/hide select option.
Like for current PR, to show `from` selector, it should click `recipient` placeholder label.
It doesn't seem to good idea to implement those all logics to helper function.

[the remark this answers is not in the document]
Comment (rrrooommmaaa) [reply test/source/browser/controllable.ts:251]: Ok, when we click on `recipient` label, `From: ` div becomes visible. Why shouldn't we test this?
All the tests that use `selectOption` became weaker in a sense that they won't fail on an update that would screw the current behavior (that is, if 'From" doesn't appear when clicking on the recipient). 
And why should the tests be able to click on option invisible to the user? We should test not only the sending, but that the user is capable of selecting this option.

**Review by rrrooommmaaa — APPROVED** [review]

**Pull request merged**